src/services/1inch.execute.tsx

```diff
@@ -55,17 +47,9 @@ export class OneInchExecutionManager {
       if (swapProcess.txHash) {
         tx = await signer.provider.getTransaction(swapProcess.txHash)
       } else {
-        const userAddress = await signer.getAddress()
-        const call = await oneInch.buildTransaction(
-          action.fromChainId,
-          action.fromToken.id,
-          action.toToken.id,
-          srcAmount.toString(),
-          userAddress,
-          destAddress,
-          action.slippage,
-        )
-        tx = await signer.sendTransaction(call)
+        const personalizedStep = await personalizeStep(signer, step)
+        const { tx: transaction } = await Lifi.getStepTransaction(personalizedStep)
+        tx = await signer.sendTransaction(transaction)
```

Comment:
I know this was one done in one process before. But since `getStepTransaction` is a backend call, where the user has to wait, and `sendTransaction` requires the user to confirm a message in his wallet I would split this up. So we first get the transaction (`swapProcess.status = 'PENDING'`) and then ask the user to sign the message (`swapProcess.status = 'ACTION_REQUIRED'`)